Mathlib/FieldTheory/Minpoly/ConjRootClass.lean

```diff
@@ -0,0 +1,190 @@
+/-
+Copyright (c) 2022 Yuyang Zhao. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yuyang Zhao
+-/
+import Mathlib.FieldTheory.Minpoly.IsConjRoot
+
+/-!
+# Conjugate root classes
+
+In this file, we define the `ConjRootClass` of a field extension `L / K` as the quotient of `L` by
+the relation `IsConjRoot K`.
+-/
+
+variable (K L S : Type*) [Field K] [Field L] [Field S]
+variable [Algebra K L] [Algebra K S] [Algebra L S] [IsScalarTower K L S]
+
+/-- `ConjRootClass K L` is the quotient of `L` by the relation `IsConjRoot K`. -/
+def ConjRootClass := Quotient (α := L) (IsConjRoot.setoid K L)
+
+namespace ConjRootClass
+
+variable {L}
+
+/-- The canonical quotient map from a field `K` into the `ConjRootClass` of the field extension
+`L / K`. -/
+def mk (x : L) : ConjRootClass K L :=
+  ⟦x⟧
+
+@[simp]
+theorem mk_def {x : L} : ⟦x⟧ = mk K x := rfl
+
+instance : Zero (ConjRootClass K L) :=
+  ⟨mk K 0⟩
+
+@[elab_as_elim, cases_eliminator, induction_eliminator]
+lemma ind {motive : ConjRootClass K L → Prop} (h : ∀ x : L, motive (mk K x))
+    (c : ConjRootClass K L) : motive c :=
+  Quotient.ind h c
+
+variable {K}
+
+@[simp]
+theorem mk_eq_mk {x y : L} : mk K x = mk K y ↔ IsConjRoot K x y := Quotient.eq
+
+@[simp]
+theorem mk_zero : mk K (0 : L) = 0 :=
+  rfl
+
+@[simp]
+theorem mk_eq_zero_iff (x : L) : mk K x = 0 ↔ x = 0 := by
+  rw [eq_comm (b := 0), ← mk_zero, mk_eq_mk, isConjRoot_zero_iff_eq_zero]
+
+instance [Normal K L] [DecidableEq L] [Fintype (L ≃ₐ[K] L)] : DecidableEq (ConjRootClass K L) :=
+  Quotient.decidableEq (d := IsConjRoot.decidable)
+
+/-- `c.carrier` is the set of conjugates represented by `c`. -/
+def carrier (c : ConjRootClass K L) : Set L :=
+  mk K ⁻¹' {c}
+
+theorem mem_carrier {x : L} {c : ConjRootClass K L} : x ∈ c.carrier ↔ mk K x = c :=
+  Iff.rfl
+
+theorem carrier_zero : (0 : ConjRootClass K L).carrier = {0} := by
```

Comment:
`mem_carrier` is used a lot in `simp` calls below, so it can usefully be made `@[simp]` I think. For `carrier_zero` I don't see any uses here, but the right hand side is certainly simpler than the left.
```suggestion
@[simp]
theorem mem_carrier {x : L} {c : ConjRootClass K L} : x ∈ c.carrier ↔ mk K x = c :=
  Iff.rfl

@[simp]
theorem carrier_zero : (0 : ConjRootClass K L).carrier = {0} := by
```